### PR TITLE
Remove root-owned files

### PR DESCRIPTION
**Pull request by kzys**
Signed-off-by: Kazuyoshi Kato <katokazu@amazon.com>

*Issue #, if available:*

*Description of changes:*

We run some tests as root user whereas BuildKite's agent is running as non-root due to security concerns. Because of the mismatch, we sometimes have files that cannot be deleted from BuildKite's agent.

This PR does the following, which is larger than what I had expected:
- Clean up files from tests correctly
- Update .buildkite/pipeline.yml to download binaries as non-root
- Update .buildkite/hooks/pre-exit to find files as root
- Remove .buildkite/hooks/pre-checkout, since the hook cannot be executed before cloning this Git repos, which is too late.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .buildkite/hooks/pre-exit [hunk 1]
```diff
@@ -1,4 +1,7 @@
 #!/bin/bash
 set -euo pipefail
 
-find . -user root -print0 | xargs -0 sudo rm -rf '{}'
+# Remove all root-owned files.
+# Since BuildKite's Agent is running as non-root. Root-owned files cannot be
+# deleted by the agent.
+sudo sh -c 'find . -user root -print0 | xargs -0 rm -rf "{}"'
```
Comment (alakesh) [inline .buildkite/hooks/pre-exit:7]: Anything with rm -rf sounds scary. Can we do it in a more controlled way? For example collect the list of file first and then delete each one of them logging them on each deletion?

Comment (kzys) [reply .buildkite/hooks/pre-exit:7]: How about `rm -rfv`? So that we can log what we've deleted.

**Comment by FFMMM** [comment]
Could we get more context as to the issue we are trying to solve here? What's creating those `root` files to begin with? Why do we need to change this now?

**Comment by kzys** [comment]
@FFMMM Sure. We run some tests as `root` user whereas BuildKite's agent is running as non-root due to security concerns. Because of the mismatch, we sometimes have files that cannot be deleted from BuildKite's agent.

**Pull request closed without merging**